pkg/wallet/wallet.go

```diff
@@ -371,7 +391,10 @@ func Import(nickname string, privateKeyB58V string, password string) (*Wallet, e
 	return wallet, nil
 }
 
-func CreateWalletFromKeys(nickname string, privateKey []byte, publicKey []byte, password string) (*Wallet, error) {
+func CreateWalletFromKeys(nickname string, privateKey ed25519.PrivateKey, password string) (*Wallet, error) {
+
+	publicKey := reflect.ValueOf(privateKey.Public()).Bytes() // force conversion to byte array
```

Comment:
great! with privateKey.Public() of package ed25519 we can use ed25519 algorithm (or curve) to create our public key :)